oak_restricted_kernel/src/lib.rs

```diff
@@ -156,6 +156,36 @@ pub fn start_kernel(info: &BootParams) -> ! {
     // Safety: in the linker script we specify that the ELF header should be placed at 0x200000.
     let program_headers = unsafe { elf::get_phdrs(VirtAddr::new(0x20_0000)) };
 
+    let _stage0_dice_data = {
+        let dice_memory_slice = {
+            let e820_dice_data_entry = info
+                .e820_table()
+                .iter()
+                .find(|e| e.entry_type() == Some(oak_linux_boot_params::E820EntryType::DiceData))
+                .expect("failed to find dice data");
+
+            // safety: the E820 table indicated that this is the corrct memory segment
+            unsafe {
+                core::slice::from_raw_parts_mut::<u8>(
+                    e820_dice_data_entry.addr() as *mut u8,
+                    e820_dice_data_entry.size(),
+                )
+            }
+        };
+        let mut dice_data: oak_dice::evidence::Stage0DiceData =
+            <oak_dice::evidence::Stage0DiceData as zerocopy::FromZeroes>::new_zeroed();
+        <oak_dice::evidence::Stage0DiceData as zerocopy::AsBytes>::as_bytes_mut(&mut dice_data)
+            .clone_from_slice(dice_memory_slice);
+
+        // overwrite the dice data provided by stage0 after reading
+        dice_memory_slice.fill(0);
+
+        if dice_data.magic != oak_dice::evidence::STAGE0_MAGIC {
+            panic!("dice data loaded from stage0 failed validation");
+        }
+        dice_data
+    };
+
     // Physical frame allocator
     mm::init(info.e820_table(), program_headers);
```

Comment:
implementing this code after page tables are initialized leads to page faults, unsure why.